custom_model_runner/datarobot_drum/drum/perf_testing.py

```diff
@@ -523,25 +523,21 @@ def validation_test(self):
 
         results = {}
 
-        for column_name in column_names:
-            with NamedTemporaryFile(
-                mode="w",
-                dir=null_datasets_dir,
-                prefix="null_value_imputation_{}_".format(column_name),
-                delete=False,
-            ) as temp_f:
-                temp_data_name = temp_f.name
-                df_tmp = df.copy()
-                df_tmp[column_name] = None
-                df_tmp.to_csv(temp_data_name, index=False)
-                CMRunnerUtils.replace_cmd_argument_value(
-                    cmd_list, ArgumentsOptions.INPUT, temp_data_name
-                )
+        for i, column_name in enumerate(column_names):
+            tmp_dataset_file_path = os.path.join(
```

Comment:
doesn't this need to be removed after it's used?

Reply:
Whole folder is removed if test passes and kept if test fails.